src/internal/_equals.js

```diff
@@ -14,10 +14,19 @@ module.exports = function _equals(a, b, stackA, stackB) {
     return false;
   }
 
+  if (a == null || b == null) {
+    return false;
+  }
+
+  if (typeof a.equals === 'function') {
+    return a.equals(b);
+  }
+
+  if (typeof b.equals === 'function') {
+    return b.equals(a);
+  }
+
```

Comment:
Lawn-mowing must have done me in.  I swear I submitted this comment last night!  :smile:

There is probably nothing we can do about this; I'm guessing its a well-known problem with some fancy name.  But I noticed it here; this sort of implementation can lead to a non-commutativity of `equals` in the face of type inheritance:

```js
var s = new Square(5);
var r = new Rectangle(5, 5);
R.equals(s, r); //=> false (r not instance of Square)
R.equals(r, s); //=> true (s instance of Rectangle, r.height (/width) == s.height (/width) )
```

Not expecting any change.  Just sayin'.